index/scorch/segment/zap/merge.go

```diff
@@ -649,7 +649,10 @@ func mergeStoredAndRemap(segments []*SegmentBase, drops []*roaring.Bitmap,
 				fieldID := int(fieldsMap[field]) - 1
 				vals[fieldID] = append(vals[fieldID], value)
 				typs[fieldID] = append(typs[fieldID], typ)
-				poss[fieldID] = append(poss[fieldID], pos)
+
```

Comment:
Hi @abhinavdangeti - not quite the same issue, but I'm thinking the value param of the callback might have the same problem (where the callback needs to copy the value)?

Reply:
I mentioned the 'value' in my comment earlier.  It does not have the same problem, because it is shared from the uncompressed slice.  But it could have a different problem, if say you closed the index and were holding on to a search result from earlier...

Reply:
Hmm, one worry is perhaps there might be a problem due to sync.Pool reuse of the visitDocumentCtx structs, where the uncompressed struct might be inadvertently reused by a concurrent goroutine.  Hmm.